src/apps/search-result/search-result.tsx

```diff
@@ -87,13 +87,8 @@ const SearchResult: React.FC<SearchResultProps> = ({ q, pageSize }) => {
   // This is an initial, intentionally simple approach supporting what is required by the search header.
   // It could be reworked to support all filters and terms at a later point.
   useEffect(() => {
-    const materialTypeUrlFilter = getUrlQueryParam("materialType");
-    if (materialTypeUrlFilter) {
-      addToFilter({
-        facet: FacetField.MaterialTypes,
-        term: { key: materialTypeUrlFilter, term: materialTypeUrlFilter }
-      });
-    }
+    addFilterFromUrlParam(FacetField.MaterialTypes);
```

Comment:
I think is a bit misconceiving to call it `addFilterFromUrlParam` because I think it implies that you perform the action right away.
Maybe it could be called  something like `addFilterFromUrlParamListener` to emphasise that it only performs an action if an url param exist.